test/functional/tests/io/test_write_fetch.py

```diff
@@ -0,0 +1,192 @@
+        check_statistics(cache, validated_stat_name, cache_line_size.value, file_size)
+    with TestRun.step("Verify number of writes to cache device using iostat. Shall be 150M "
+                      "(+ metadata for WB)."):
+        check_io_stats(cache_disk, cache, io_stats_before_io, file_size)
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+@pytest.mark.parametrize("cache_mode", [CacheMode.WO, CacheMode.WB, CacheMode.WT])
+@pytest.mark.parametrize("cache_line_size", CacheLineSize)
+def test_write_fetch_partial_misses(cache_mode, cache_line_size):
+    """
+        title: No caching of partial write miss operations
+        description: |
+          Validate Intel CAS ability to not cache entire cache line size for
+          partial write miss operations
+        pass_criteria:
+          - Appropriate number of write partial misses, write hits and writes to cache
+            in cache statistics
+          - Appropriate number of writes to cache in iostat
+    """
+    pattern = f"0x{uuid.uuid4().hex}"
+    file_size = Size(600, Unit.MebiByte)
+    validated_stat_name = "write partial misses"
+
+    with TestRun.step("Prepare devices."):
+        cache_disk = TestRun.disks['cache']
+        core_disk = TestRun.disks['core']
+        core_disk.create_partitions([file_size + Size(1, Unit.MebiByte)])
+        core_part = core_disk.partitions[0]
+
+    with TestRun.step("Fill core partition with pattern."):
+        cache_mode_traits = CacheMode.get_traits(cache_mode)
+        if CacheModeTrait.InsertRead in cache_mode_traits:
+            run_fio(core_part.system_path,
+                    ReadWrite.write.name,
+                    Size(4, Unit.KibiByte),
+                    file_size,
+                    True,
+                    pattern)
+        else:
+            TestRun.LOGGER.info(f"Skipped for {cache_mode} cache mode.")
+
+    with TestRun.step("Start cache and add core."):
+        cache = casadm.start_cache(cache_disk, cache_mode, cache_line_size)
+        Udev.disable()
+        core = cache.add_core(core_part)
+    with TestRun.step("Cache half of file."):
+        run_fio(core.system_path,
+                f"{'read' if CacheModeTrait.InsertRead in cache_mode_traits else 'write'}:"
+                f"{int(cache_line_size.value.get_value(Unit.KibiByte))}k",
+                cache_line_size.value,
+                file_size,
+                True,
+                pattern)
+        if CacheModeTrait.InsertRead not in cache_mode_traits:
+            cache.flush_cache()
+        casadm.reset_counters(cache.cache_id, core.core_id)
+    with TestRun.step("Run writes to CAS device using fio."):
+        io_stats_before_io = cache_disk.get_io_stats()
+        run_fio(core.system_path,
+                f"write:{int((cache_line_size.value / 2).get_value(Unit.KibiByte))}k",
```

Comment:
Please make operation type separate parameter from its size. For setting operation type use values from enum. Operation size could be type of Size() instead of string